realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -115,6 +136,105 @@ void notifySessionError(int errorCode, String errorMessage) {
         }
     }
 
+    // Called from native code
+    @SuppressWarnings("unused")
+    @KeepMember
+    synchronized void notifyProgressListener(long listenerId, long transferredBytes, long transferableBytes) {
+        Pair<ProgressListener, Progress> listener = listenerIdToProgressListenerMap.get(listenerId);
+        if (listener != null) {
+            Progress newProgressNotification = new Progress(transferredBytes, transferableBytes);
+            if (!newProgressNotification.equals(listener.second)) {
+                listener.first.onChange(newProgressNotification);
+                listener.second = newProgressNotification;
```

Comment:
Good catch. Yeah, I should assign it first. We catch exceptions in the method calling this.